### PR TITLE
feature/java moving nongen

### PR DESCRIPTION
**Pull request by baywet**
- adds implementation for IUploadSession in java
- renames graph service client following moves into core
- removes the graph service client interface to avoid shipping breaking changes
- sets a lean builder for the graph client
- removes java extensions/generated folders
- adds missing comments
- adds version provider for client

## Summary

This pull request reflects the generation changes to support the move of non-generated code into the core library
related generation difference https://github.com/microsoftgraph/msgraph-sdk-java/pull/615

### CODE REVIEW

File: Templates/templates/Java/models_extensions/BaseEntity.java.tt [hunk 1]
```diff
@@ -40,4 +40,26 @@ import java.util.HashMap;
 <#=TypeHelperJava.CreatePropertyDef(((OdcmClass)c).Properties, c is OdcmComplexClass)#>
 <#=TypeHelperJava.CreateRawJsonObject()#>
 <#=TypeHelperJava.UpdatePropertiesWithinSetRawObject(((OdcmClass)c).Properties, c is OdcmComplexClass)#>
+<# if(c.TypeName().Equals("UploadSession")) {#>
```
Comment (MIchaelMainer) [inline Templates/templates/Java/models_extensions/BaseEntity.java.tt]: Where was IUploadSession implementation before this generated change?

File: Templates/templates/Java/requests_extensions/BaseClient.java.tt [hunk 2]
```diff
@@ -5,33 +5,15 @@
...
-            endpoint = DEFAULT_GRAPH_ENDPOINT;
-        }
-        return endpoint;
-    }
-
-    @Override
-    public void setServiceRoot(@Nonnull final String value) {
-        endpoint = value;
+    protected GraphServiceClient() {
+        setServiceRoot("<#=ConfigurationService.Settings.DefaultBaseEndpointUrl#>");
```
Comment (MIchaelMainer) [inline Templates/templates/Java/requests_extensions/BaseClient.java.tt]: Is setServiceRoot now on a base class?

Comment (MIchaelMainer) [reply Templates/templates/Java/requests_extensions/BaseClient.java.tt]: Looking at another commits appears to address my question

**Review by MIchaelMainer — COMMENTED** [review]
Reviewed commit by commit. Too many renames for easy local review.

File: Templates/templates/Java/models_extensions/BaseEntity.java.tt (hunk 1 above)
Comment (baywet) [reply Templates/templates/Java/models_extensions/BaseEntity.java.tt]: before this generated change the UploadSession type was used directly by the provider. We didn't have an interface.

File: Templates/templates/Java/requests_extensions/BaseClient.java.tt (hunk 2 above)
Comment (baywet) [reply Templates/templates/Java/requests_extensions/BaseClient.java.tt]: yes, I moved the method into the base type in core.

**Review by MIchaelMainer — APPROVED** [review]

File: scripts/clean-java-files.ps1 [hunk 3]
```diff
@@ -1,8 +1,8 @@
-$extensionsAndGeneratedDirectories = Get-ChildItem $env:MainDirectory -Include extensions,generated -Recurse -Directory
+$modelsAndRequestsDirectories = Get-ChildItem $env:MainDirectory -Include models,requests -Recurse -Directory
 
-# this list should be updated if a new hand-crafted extension is added to one of the extensions/ directories
-$filesThatShouldNotBeDeleted = "UploadSession.java","DateOnly.java","TimeOfDay.java","Multipart.java","ChunkedUploadRequest.java","ChunkedUploadResult.java","CustomRequestBuilder.java"
-foreach ($directory in $extensionsAndGeneratedDirectories)
+# this list should be updated if a new hand-crafted extension is added to one of the directories
+$filesThatShouldNotBeDeleted = @()
```
Comment (zengin) [inline scripts/clean-java-files.ps1:4]: remove this parameter and the -Exclude flag below.

**Review by zengin — APPROVED** [review]

File: scripts/clean-java-files.ps1 (hunk 3 above)
Comment (baywet) [reply scripts/clean-java-files.ps1:4]:
```suggestion
```

File: scripts/clean-java-files.ps1 [hunk 4]
```diff
@@ -1,8 +1,8 @@
...
+foreach ($directory in $modelsAndRequestsDirectories)
 {
     Remove-Item $directory.FullName -Recurse -Force -Exclude $filesThatShouldNotBeDeleted
```
Comment (baywet) [inline scripts/clean-java-files.ps1:7]:
```suggestion
    Remove-Item $directory.FullName -Recurse -Force
```

File: scripts/clean-java-files.ps1 [hunk 5]
```diff
@@ -1,8 +1,8 @@
-$extensionsAndGeneratedDirectories = Get-ChildItem $env:MainDirectory -Include extensions,generated -Recurse -Directory
+$modelsAndRequestsDirectories = Get-ChildItem $env:MainDirectory -Include models,requests -Recurse -Directory
 
-# this list should be updated if a new hand-crafted extension is added to one of the extensions/ directories
-$filesThatShouldNotBeDeleted = "UploadSession.java","DateOnly.java","TimeOfDay.java","Multipart.java","ChunkedUploadRequest.java","ChunkedUploadResult.java","CustomRequestBuilder.java"
-foreach ($directory in $extensionsAndGeneratedDirectories)
+# this list should be updated if a new hand-crafted extension is added to one of the directories
```
Comment (baywet) [inline scripts/clean-java-files.ps1:3]:
```suggestion
```

**Pull request merged**